web-client/src/ustc-ui/InlineLink/InlineLink.tsx

```diff
@@ -0,0 +1,41 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
+import React from 'react';
+import classNames from 'classnames';
+
+export const InlineLink = ({
+  children,
+  className = '',
+  href,
+  icon,
+  iconSize = '1x',
+  tooltip,
+}: {
+  children: string | React.ReactNode;
+  className?: string;
+  icon?: IconProp;
+  href: string;
+  iconSize?: '1x' | '2x' | '3x' | '4x' | '5x';
+  tooltip?: string;
```

Comment:
Might be worth moving out as a separate interface for readability, but no big deal either way.

Reply:
I'm personally not all that opinionated on whether inline types or an extracted type is more readable in this file. Extracting the type though does mean you lose the more granular type information on hover where this component is referenced. E.g. if I make this "InlineLinkProps" then hover over a use of InlineLink, all I see is...
![Screenshot 2024-09-18 at 8 34 51 AM](https://github.com/user-attachments/assets/dd6e5a01-ca06-4fe2-b77a-9b2220ee7364)

But if I keep it inline I get...
![Screenshot 2024-09-18 at 8 34 25 AM](https://github.com/user-attachments/assets/a9b79d98-76e9-423a-ab58-9a84d3600efb)

I could see how this becomes a potential downside for a component with many props, but for one with relatively few it seems a decent QoL improvement. Happy to chat more if you have other feels or if I'm missing anything 😄

Reply:
I'm happy either way, and you have good reasons here :)